ring/ring_test.go

```diff
@@ -627,27 +628,31 @@ func TestRing_Get_ZoneAwareness(t *testing.T) {
 		"should succeed if there are enough instances per zone on RF = 3": {
 			numInstances:         16,
 			numZones:             3,
+			totalZones:           3,
 			replicationFactor:    3,
 			zoneAwarenessEnabled: true,
 			expectedInstances:    3,
 		},
 		"should fail if there are instances in 1 zone only on RF = 3": {
```

Comment:
Without the change to add an instance in each zone below, this test would fail.

Previously, `findInstancesForKey` ignored the actual number of zones, `len(r.ringTokensByZone)`, and assumed that replication factor was equivalent. Now that we've changed things so that replication factor is not necessarily the same as the number of zones, we need to add at least one instance in each zone so that `findInstancesForKey` is aware of them and only picks a single instance from each zone.

Reply:
hm, in this case then dynamic replication only works for zone-aware replication. Should we add this to the docs and validation code?

Reply:
> hm, in this case then dynamic replication only works for zone-aware replication. Should we add this to the docs and validation code?

It still works for non zone-aware replication. It just needs instances with tokens to determine the number of zones, even if the number of zones is one.

Reply:
I've since changed how `findInstancesForKey` works such that this isn't required anymore. Now, we explicitly treat the configured replication factor as the number of zones when zone-awareness is enabled.